src/prog_models/models/centrifugal_pump.py

```diff
@@ -121,79 +122,92 @@ class CentrifugalPump(prognostics_model.PrognosticsModel):
-        Trdot = 1/self.parameters['mcRadial'] * (x['rRadial']*x['w']*x['w'] - self.parameters['HRadial1']*(x['Tr']-u['Tamb']) - self.parameters['HRadial2']*(x['Tr']-x['To']))
+        params = self.parameters
+        Todot = 1/params['mcOil'] * (params['HOil1']*(x['Tt']-x['To']) + params['HOil2']*(x['Tr']-x['To'])\
+            + params['HOil3']*(u['Tamb']-x['To']))
+        Ttdot = 1/params['mcThrust'] * (x['rThrust']*x['w']*x['w'] - params['HThrust1']*(x['Tt']-u['Tamb'])\
+            - params['HThrust2']*(x['Tt']-x['To']))
+        Adot = -params['wA']*x['Q']*x['Q']
+        rRadialdot = params['wRadial']*x['rRadial']*x['w']*x['w']
+        rThrustdot = params['wThrust']*x['rThrust']*x['w']*x['w']
+        friction = (params['r']+x['rThrust']+x['rRadial'])*x['w']
+        QLeak = math.copysign(params['cLeak']*params['ALeak']*math.sqrt(abs(u['psuc']-u['pdisch'])), \
+            u['psuc']-u['pdisch'])
+        Trdot = 1/params['mcRadial'] * (x['rRadial']*x['w']*x['w'] - params['HRadial1']*(x['Tr']-u['Tamb']) - params['HRadial2']*(x['Tr']-x['To']))
         slipn = (u['wsync']-x['w'])/(u['wsync'])
-        ppump = x['A']*x['w']*x['w'] + self.parameters['b']*x['w']*x['Q']
+        ppump = x['A']*x['w']*x['w'] + params['b']*x['w']*x['Q']
         Qout = max(0,x['Q']-QLeak)
         slip = max(-1,(min(1,slipn)))
         deltaP = ppump+u['psuc']-u['pdisch']
-        Te = 3*self.parameters['R2']/slip/(u['wsync']+0.00001)*u['V']**2/((self.parameters['R1']+self.parameters['R2']/slip)**2+(u['wsync']*self.parameters['L1'])**2)
-        backTorque = -self.parameters['a2']*Qout**2 + self.parameters['a1']*x['w']*Qout + self.parameters['a0']*x['w']**2
-        Qo = math.copysign(self.parameters['c']*math.sqrt(abs(deltaP)), deltaP)
-        wdot = (Te-friction-backTorque)/self.parameters['I']
-        Qdot = 1/self.parameters['FluidI']*(Qo-x['Q'])
-        QLeak = math.copysign(self.parameters['cLeak']*self.parameters['ALeak']*math.sqrt(abs(u['psuc']-u['pdisch'])), u['psuc']-u['pdisch'])
-        return self.apply_process_noise({
-            'A': x['A'] + Adot*dt,
-            'Q': x['Q'] + Qdot*dt, 
-            'To': x['To'] + Todot*dt,
-            'Tr': x['Tr'] + Trdot*dt,
-            'Tt': x['Tt'] + Ttdot*dt,
-            'rRadial': x['rRadial'] + rRadialdot*dt,
-            'rThrust': x['rThrust'] + rThrustdot*dt,
-            'w': x['w']+wdot*dt,
-            'wA': x['wA'],
-            'wRadial': x['wRadial'],
-            'wThrust': x['wThrust'],
+        Te = 3*params['R2']/slip/(u['wsync']+0.00001)*u['V']**2 \
+            /((params['R1']+params['R2']/slip)**2+(u['wsync']*params['L1'])**2)
+        backTorque = -params['a2']*Qout**2 + params['a1']*x['w']*Qout + params['a0']*x['w']**2
+        Qo = math.copysign(params['c']*math.sqrt(abs(deltaP)), deltaP)
+        wdot = (Te-friction-backTorque)/params['I']
+        Qdot = 1/params['FluidI']*(Qo-x['Q'])
```

Comment:
From the proceeding paper, it looks like the sign may be flipped here (i.e. x['Q']-Qo instead of Qo-x['Q']):

Daigle, M. J., & Goebel, K. (2013). Model-based prognostics with concurrent damage progression processes. IEEE Transactions on Systems, Man, and Cybernetics: Systems, 43(3), 535–546. https://doi.org/10.1109/tsmca.2012.2207109 

Reply:
Currently investigating this one. The implemented solution is consistent with the matlab implementation as well. Currently I suspect this might be a mistake in the paper, but we're working to confirm.